File: LeetCode/55. Jump Game/elixir/solution.exs
defmodule Solution do
  @spec can_jump(nums :: [integer]) :: boolean
  def can_jump([head | rest] = _nums), do: jump(rest, head - 1)

  defp jump([], _reach), do: true
  defp jump(_nums, reach) when reach < 0, do: false
  defp jump([h | t], reach), do: jump(t, max(h, reach) - 1)
end

File: LeetCode/55. Jump Game/java/Solution.java
class Solution {
    public boolean canJump(int[] nums) {
        int reach = nums[0];
        for (int i = 1; i < nums.length; i++) {
            if (reach < i) {
                return false;
            } else {
                reach = Math.max(reach, i + nums[i]);
            }
        }
        return true;
    }
}